winch/codegen/src/isa/x64/mod.rs

```diff
@@ -113,4 +119,15 @@ impl TargetIsa for X64 {
         // See `cranelift_codegen`'s value of this for more information.
         16
     }
+
+    fn host_to_wasm_trampoline(&self, ty: &FuncType) -> Result<MachBufferFinalized<Final>> {
```

Comment:
In this case, are we talking about the system ABI or the new Winch internal ABI?

Reply:
Actually, maybe I'm misunderstanding but I think we are saving registers in the trampoline already? I take all the arguments passed to the trampoline and push them onto the stack prior to calling the passed function. After that, I load values back into registers as I need them.

Reply:
Yeah, but we also need to save all the other callee-saved registers, not just the ones passed in as params, depending on what the system ABI considers callee-saved. For example: if r12 is callee saved, we need to save it (it's callee saved under the SystemV ABI) even if it's not used as a argument register because we are going to potentially clobber it.

Reply:
> In this case, are we talking about the system ABI or the new Winch internal ABI?

In the case of trampolines it's the system ABI that one that we are referring to.

Reply:
Ah yes that makes sense! I can add a note for this